cli/src/commands/util.rs

```diff
@@ -110,12 +107,14 @@ fn cmd_util_completion(
 ) -> Result<(), CommandError> {
     let mut app = command.app().clone();
     let mut buf = vec![];
-    let shell = if args.zsh {
-        clap_complete::Shell::Zsh
-    } else if args.fish {
-        clap_complete::Shell::Fish
-    } else {
-        clap_complete::Shell::Bash
+    let shell = match (args.shell, args.fish, args.zsh, args.bash) {
+        (Some(s), false, false, false) => s,
+        // allow `--fish` and `--zsh` for back-compat, but don't allow them to be combined
+        (None, true, false, false) => Shell::Fish,
+        (None, false, true, false) => Shell::Zsh,
+        // default to bash for back-compat. TODO: consider making `shell` a required argument
+        (None, false, false, _) => Shell::Bash,
+        _ => return Err(user_error("cannot generate completion for multiple shells at once")),
```

Comment:
I found the old code a bit more readable.

Reply:
hmm, the old code didn't exhaustively match. i can change it, but then i'd need to explicitly check for any two of shell/fish/zsh/bash being set, which is non-trivial.

Reply:
As we'll need to add a warning message for each option though, I'd prefer to keep it as simple as possible. The option to revisit is always open.

Reply:
i really don't think it will be more readable the other way. can you take a look at the current code and see if you like it any better? it warns on each option now.

Reply:
good enough.